compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/TreeModifier.java

```diff
@@ -3070,14 +3070,23 @@ protected Node transformSyntaxNode(Node node) {
         }
 
         boolean nodeModified = false;
-        STNode[] newSTNodes = new STNode[nodeList.size()];
+        STNode[] newSTNodes = new STNode[nodeList.internalListNode.size()];
+
         for (int index = 0; index < nodeList.size(); index++) {
             T oldNode = nodeList.get(index);
             T newNode = modifyNode(oldNode);
             if (oldNode != newNode) {
                 nodeModified = true;
             }
-            newSTNodes[index] = newNode.internalNode();
+
+            if (nodeList instanceof SeparatedNodeList && index < nodeList.size() - 1) {
+                newSTNodes[2 * index] = newNode.internalNode();
+                newSTNodes[(2 * index) + 1] = ((SeparatedNodeList<?>) nodeList).getSeparator(index).internalNode();
```

Comment:
Looks like the separator is not modified.
Shall we add a test case to check whether the separator is also modified? e.g: remove all whitespaces

Reply:
alright

Reply:
ok